website/src/blogs/2017-06-29-storage-infrastructure-as-code-using-openebs.md

```diff
@@ -0,0 +1,79 @@
+
+So, let us see what are the areas in which storage as infrastructure has to be thought in terms of code? This can be answered using the angle of who interacts with storage or the user persona. The simple answer would be the application developer and storage platform engineer. Next, what are the challenges today for a developer of a stateful application and the developer or engineer of underlying persistent storage platform? In short, the developer needs
+
+- Some amount of persistent storage
+- With guaranteed availability
+- With guaranteed latency
+
+The developer needs to quantify the amount of storage and the latency levels, which is a challenge always. The developer always starts with a guesstimate and refines later. The guaranteed availability is a characteristic of the underlying storage platform, but the developer may still need to decide some aspects related to the availability such as “how many copies of the data is needed, how often backups are to be taken, etc”. All these decisions to be taken by the developer, bring in the need for storage expertise or knowledge to the application developer, which usually is an additional burden to the developer. Traditionally, these decisions are taken by storage administrators who are adjacent to the application developers, creating a gap between what is needed and what is available. As the application scales, this gap widens, the application fails. **Yes, because something went wrong with storage !!**
+
+The solution to this well known and well understood problem is to move towards a system where the application developer auto-magically gets help from the underlying storage system and he/she can continue to express/code the storage needs through the YAML language. The “so-called” storage administrator in the enterprise is going to be helped with an intelligent-storage-robot working continuously to align the storage to the application needs. With OpenEBS, the “intelligent-storage-robot” comes as an inherent component of the storage platform. The storage administrator can scale the platform and manage it at ease and do the job very well.
+
+The intelligent-storage-robot or what we call OpenEBS as “Storage bot” handles the tough job of automating the storage actions such as
+
+- Updating the storage catalogues for application developers
+- Work with the storage platform to identify the capacity change needs, migration needs, backup needs and coordinate with the application
+- Work with the application for scheduling application aware snapshots, capacity management, data migration changes, automated data recovery etc
+- Work with a centralised ML engine (if there is one) to upload the application behaviour or to learn about the application behaviour so that it becomes more intelligent
```

Comment:
```suggestion
- Work with a centralized ML engine (if there is one) to upload the application behaviour or to learn about the application behaviour so that it becomes more intelligent
```